crates/client/mempool/src/lib.rs

```diff
@@ -124,11 +124,13 @@ impl Mempool {
         } else {
             None
         };
-
         // Perform validations
         let exec_context = ExecutionContext::new_in_block(Arc::clone(&self.backend), &pending_block_info)?;
         let mut validator = exec_context.tx_validator();
-        let _ = validator.perform_validations(clone_account_tx(&tx), deploy_account_tx_hash.is_some());
+
+        if let Transaction::AccountTransaction(account_tx) = clone_transaction(&tx) {
+            let _ = validator.perform_validations(account_tx, deploy_account_tx_hash.is_some());
+        }
 
         if !is_only_query(&tx) {
```

Comment:
probably not for this PR but shouldn't we throw an error if someone calls `add_invoke_transaction` as a query? query versions should only be allowed in `estimate_fee` and `estimate_message_fee` imo. so if it's a query version on a write call we can revert much earlier maybe?